skcosmo/SparseMethods/SparseKPCA.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+from .Sparsified import _Sparsified
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils import check_array
+from ..utils.eig_solver import eig_solver
+from sklearn.exceptions import NotFittedError
+from ..selection.FPS import SampleFPS
+
+
+class SparseKPCA(_Sparsified):
+    """
+    ----Inherited Attributes----
+    :param kernel: the kernel used for this learning
+    :param gamma: exponential factor of the rbf and sigmoid kernel
+    :param degree: polynomial kernel degree
+    :param coef0: free term of the polynomial and sigmoid kernel
+    :param kernel_params: kernel parameter set
+    :param n_active: the size of the small dataset used in learning
+    :param tol: Relative accuracy for eigenvalues (stopping criterion) The default value of 0 implies machine precision.
+    :param center: if True, centering is carried out
+    :param n_jobs: The number of jobs to use for the computation the kernel. This works by breaking down the pairwise matrix into n_jobs even slices and computing them in parallel.
+    :param n_components: number of components for which selection is carried out in SparseKPCA and SparseKRCovR
+    """
+
+    def __init__(
+        self,
+        n_components,
+        kernel="linear",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        kernel_params=None,
+        n_active=None,
+        tol=0,
+        center=True,
+        n_jobs=1,
+        selector=SampleFPS,
+        fit_inverse_transform=False,
+        copy_X=True,
+    ):
+        super().__init__(
+            kernel=kernel,
+            gamma=gamma,
+            degree=degree,
+            coef0=coef0,
+            kernel_params=kernel_params,
+            center=center,
+            n_active=n_active,
+            n_jobs=n_jobs,
+            selector=selector,
+        )
+        self.n_components = n_components
+        self.tol = tol
+        self.pkt_ = None
```

Comment:
You don't need to set to None

Reply:
Function `check_is_fitted(self, ["pkt_", "X_sparse_"])`  into the `transform` check `pkt_`. As I recall, unless it is explicitly equated to None, this check will not work. I can look this again.

Reply:
It should just check if it isn't set, no? Classes can get_attr, and if the attribute isn't set, it will set off the flag.